main.go

```diff
@@ -89,11 +79,18 @@ func main() {
 		MacaroonHex: os.Getenv("MACAROON_HEX"),
 	})
 	if err != nil {
-		log.Fatalf("Failed to create LND client: %v", err)
+		fmt.Printf("Failed to create LND client: %v", err)
+	}
+	lnurlClient := &lnurl.LNURLWrapper{
+		Address: os.Getenv("LNURL_ADDRESS"),
+	}
+
+	svc := &service.Service{
+		LndClient:   lndClient,
+		LnurlClient: lnurlClient,
```

Comment:
I think it would be better if we had a configuration parameter that indicates if you want to use lnd or lnurl. And then your client (which is an “interface”) is created depending on that configuration parameter. So you don’t need to write any code like “if LND is not present or there was an error, use LNURL”. If you go this way and you want to add another way of generating invoices (eg. C-Lightning), this can quickly get out of hand.

To see an example on how to integrate an interface in the `service.Service`, take a look at https://github.com/getAlby/lndhub.go/blob/main/lib/service/service.go#L22